src/acl.c

```diff
@@ -1291,7 +1293,6 @@ int ACLSetUser(user *u, const char *op, ssize_t oplen) {
         if (server.acl_pubsub_default & SELECTOR_FLAG_ALLCHANNELS)
             serverAssert(ACLSetUser(u,"allchannels",-1) == C_OK);
         serverAssert(ACLSetUser(u,"off",-1) == C_OK);
-        serverAssert(ACLSetUser(u,"sanitize-payload",-1) == C_OK);
```

Comment:
i think the bug is actually in the default newly created user, not in the reset command.
same as `off` and `on` these two bits are mutually exclusive, if the default created user doesn't have any of them set, then that's the bug.

i think i had to add USER_FLAG_SANITIZE_PAYLOAD to ACLCreateUser, where it sets flags to USER_FLAG_DISABLED.

this bug might not have any real implications, since the code in rdb.c checks for USER_FLAG_SANITIZE_PAYLOAD_SKIP, so the fact that USER_FLAG_SANITIZE_PAYLOAD is missing doesn't really matters.
please ack.

Reply:
> same as off and on these two bits are mutually exclusive, if the default created user doesn't have any of them set, then that's the bug.

yes, it make sense.

> this bug might not have any real implications, since the code in rdb.c checks for USER_FLAG_SANITIZE_PAYLOAD_SKIP, so the fact that USER_FLAG_SANITIZE_PAYLOAD is missing doesn't really matters.

ack. i checked the code, it is right